### PR TITLE
Issue #558 add matplotlib example

### PR DESCRIPTION
**Pull request by AbisoyeOnanuga**
Hello @jrobinAV if this is still open for PRs

This pull request adds a new example demonstrating how to integrate Matplotlib scatter plots within the Taipy framework using markdown syntax. The example showcases the creation of a scatter plot representing employee performance vs. happiness, with bubble sizes indicating efficiency. 

- The legend displays efficiency percentages. 
- Custom CSS is applied to ensure a responsive and well-aligned layout.

image

### CODE REVIEW

File: doc/gui/examples/charts/matplotlib/styles.css
Comment (FredLL-Avaiga) [inline doc/gui/examples/charts/matplotlib/styles.css:1]: No need for a separate file, since Taipy 4.0 you can define the style as a dict at the page level. 
ie page = Markdown("...", style = { ...css...}

File: doc/gui/examples/charts/matplotlib/matplotlib_scatter.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ax.add_artist(legend1)
+
+# Adjust layout to ensure everything fits
+plt.tight_layout(rect=[0, 0, 0.75, 1])  # Adjust the rect parameter to leave space for the legend
+
+# Define Taipy page content
+page = """
+<|toggle|theme|>
+
+<center style="color: var(--taipy-primary); font-size: 2.5em; font-weight:bold">
```
Comment (FredLL-Avaiga) [inline doc/gui/examples/charts/matplotlib/matplotlib_scatter.py]: Keep it simple and avoid complex HTML/markdown attributes that clutter the main message

File: doc/gui/examples/charts/matplotlib/matplotlib_scatter.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  **Matplotlib**{: .color-primary} 2D Scatter Plot
+</center>
+This page contains a 2D scatter plot created with Matplotlib: 
+<br/><br/>
+
+<|part|content={fig}|>
```
Comment (FredLL-Avaiga) [inline doc/gui/examples/charts/matplotlib/matplotlib_scatter.py]: Use a specific classname

File: doc/gui/examples/charts/matplotlib/styles.css [hunk 3]
```diff
@@ -0,0 +1,11 @@
+img {
+    display: block;
+    margin: auto;
+}
+
+div.taipy-part.MuiBox-root {
```
Comment (FredLL-Avaiga) [inline doc/gui/examples/charts/matplotlib/styles.css]: Reuse the classname defined as a `part` property

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
Tx, great work

A few changes

Could you add a screenshot of the result as a comment to the PR?

**Comment by AbisoyeOnanuga** [comment]
Thanks for reviewing, I'll make these updates @FredLL-Avaiga

File: doc/gui/examples/charts/matplotlib/matplotlib_scatter.py (hunk 1 above)
Comment (AbisoyeOnanuga) [reply doc/gui/examples/charts/matplotlib/matplotlib_scatter.py]: fixed

File: doc/gui/examples/charts/matplotlib/styles.css
Comment (AbisoyeOnanuga) [reply doc/gui/examples/charts/matplotlib/styles.css:1]: updating my taipy to 4.0.0
I'm unable to test right now for some connection issue when making the update, but I'll post the changes

**Comment by FredLL-Avaiga** [comment]
and please fix the linter issues

File: doc/gui/examples/charts/matplotlib_scatter.py [hunk 4]
```diff
@@ -64,17 +64,28 @@
 plt.tight_layout(rect=[0, 0, 0.75, 1])  # Adjust the rect parameter to leave space for the legend
 
 # Define Taipy page content
-page = """
-<|toggle|theme|>
+page = Markdown(
+  """
+  <|toggle|theme|>
+  # **Matplotlib** 2D Scatter Plot
```
Comment (FredLL-Avaiga) [inline doc/gui/examples/charts/matplotlib_scatter.py]: even those are not necessary

Comment (AbisoyeOnanuga) [reply doc/gui/examples/charts/matplotlib_scatter.py]: no worries

File: doc/gui/examples/charts/matplotlib_scatter.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+          "height": "560px",
+          "display": "flex",
+          "align-items": "center",
+          "justify-content": "center"
+      }
+  }
+)
+
+if __name__ == "__main__":
+    Gui(page, style="style").run(title="Chart-Scatter-Matplotlib")
```
Comment (FredLL-Avaiga) [inline doc/gui/examples/charts/matplotlib_scatter.py]: `style="style" `is not needed any more

Comment (AbisoyeOnanuga) [reply doc/gui/examples/charts/matplotlib_scatter.py]: removed

**Comment by AbisoyeOnanuga** [comment]
@FredLL-Avaiga below is the screenshot of the most recent commit to the PR after my update to Taipy 4.0.0 as of now 
image

**Comment by FredLL-Avaiga** [comment]
Why did you change the part for an image?

File: doc/gui/examples/charts/matplotlib/matplotlib_scatter.py
Comment (FredLL-Avaiga) [inline doc/gui/examples/charts/matplotlib/matplotlib_scatter.py:1]: can you rename to matplotlib_image.py ?
and then we'll merge it

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
almost there tx @AbisoyeOnanuga 
We'll merge this as soon as the file is renamed

File: doc/gui/examples/charts/matplotlib/matplotlib_scatter.py
Comment (AbisoyeOnanuga) [reply doc/gui/examples/charts/matplotlib/matplotlib_scatter.py:1]: Okay. Let me know if there's any change you want for the name or folder now that we are no longer using image

**Comment by AbisoyeOnanuga** [comment]
> Why did you change the part for an image?

I had issues with it after updating my Taipy, but I have since switched back to using part and it works and there's now no need to read any image output. Direct insertion is supported

**Comment by AbisoyeOnanuga** [comment]
screenshot of the most recent update
image

**Review by FredLL-Avaiga — DISMISSED** [review]
Tx a lot

File: doc/gui/examples/charts/matplotlib/matplotlib_image.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+                     loc="center right", bbox_to_anchor=(-0.2, 0.5), frameon=True, fontsize=10, labelspacing=2)
+ax.add_artist(legend1)
+
+# Adjust layout to ensure everything fits
+plt.tight_layout(rect=[0, 0, 0.75, 1])
+
+# Define Taipy page content
+page_content = Markdown("""
+    # Matplotlib 2D Scatter Plot
+    <|part|content={fig}|class_name=scatter-plot|>
```
Comment (FredLL-Avaiga) [inline doc/gui/examples/charts/matplotlib/matplotlib_image.py]: we'll stick with image and not part now 
Please and sorry

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]

File: doc/gui/examples/charts/matplotlib/matplotlib_image.py (hunk 6 above)
Comment (AbisoyeOnanuga) [reply doc/gui/examples/charts/matplotlib/matplotlib_image.py]: no worries,
## Renamed markdown.py to markdown_example.py to avoid circular import issue with the markdown package

# Note for Reviewers
`There is a minor issue where the colormap shades for points 1, 2, and 3 on the graph don't match exactly with those in the legend. This could be due to how Matplotlib handles colormap normalization. The labels help identify the corresponding points, mitigating any potential confusion. Despite this, the functionality remains intact and the graph is informative. `

If you encounter any suggestions or fixes for this, please feel free to contribute!

screenshot
image

File: doc/gui/examples/charts/matplotlib/markdown_example.py
Comment (FredLL-Avaiga) [inline doc/gui/examples/charts/matplotlib/markdown_example.py:1]: why did you rename that file ?

**Review by FredLL-Avaiga — APPROVED** [review]

File: doc/gui/examples/charts/matplotlib/matplotlib_image.py
Comment (FredLL-Avaiga) [inline doc/gui/examples/charts/matplotlib/matplotlib_image.py:1]: looks good

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]

File: doc/gui/examples/charts/matplotlib/matplotlib_image.py
Comment (FredLL-Avaiga) [reply doc/gui/examples/charts/matplotlib/matplotlib_image.py:1]: can you fix the linter check ?

File: doc/gui/examples/charts/matplotlib/markdown_example.py
Comment (AbisoyeOnanuga) [reply doc/gui/examples/charts/matplotlib/markdown_example.py:1]: after the merge and updating my local repo with those recent example files in the same folder. If I ran `python matplitlib_image.py` I get this circular import error: "  File "D:\Documents\vscode\taipy\doc\gui\examples\charts\matplotlib\markdown.py", line 23, in <module>
    from taipy.gui import Gui, Markdown
ImportError: cannot import name 'Gui' from partially initialized module 'taipy.gui' (most likely due to a circular import) (C:\Python311\Lib\site-packages\taipy\gui\__init__.py)" so what worked was to rename that file to markdown_example.py

That fixes the issue for me and I can then run or test my file while working on it.

File: doc/gui/examples/charts/matplotlib/matplotlib_image.py
Comment (AbisoyeOnanuga) [reply doc/gui/examples/charts/matplotlib/matplotlib_image.py:1]: okay. I made another commit

**Comment by AbisoyeOnanuga** [comment]
Hello @FredLL-Avaiga , I noticed that the linter error may still persist . This PR is a first for me i.e. contributing in Python and contributing for OpenSource, so it's hard to know what the issue was. I would be interested in learning from this. Thank you

File: doc/gui/examples/charts/matplotlib/markdown_example.py
Comment (FredLL-Avaiga) [reply doc/gui/examples/charts/matplotlib/markdown_example.py:1]: For consistency sake, I'd rather you rename the file back to markdown.py

**Comment by FredLL-Avaiga** [comment]
> Hello @FredLL-Avaiga , I noticed that the linter error may still persist . This PR is a first for me i.e. contributing in Python and contributing for OpenSource, so it's hard to know what the issue was. I would be interested in learning from this. Thank you

you can set the imports exactly as 

```py
import os

import matplotlib.pyplot as plt
from taipy.gui import Gui, Markdown

```
that should fix the linter sort imports issue

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]

File: doc/gui/examples/charts/matplotlib/matplotlib_image.py
Comment (FredLL-Avaiga) [inline doc/gui/examples/charts/matplotlib/matplotlib_image.py:1]: maybe simply rename to image.py would be enough as we are in a matplotlib folder

Comment (AbisoyeOnanuga) [reply doc/gui/examples/charts/matplotlib/matplotlib_image.py:1]: Thank you for the prompt response and sharing the import sorting snippet. I've updated these changes from the `import sort` to renaming the files to `markdown.py` and `image.py`

File: doc/gui/examples/charts/matplotlib/image.py [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+# To execute this script, make sure that the taipy-gui package is installed in your
+# Python environment and run:
+#     python <script>
+# -----------------------------------------------------------------------------------------
+import os
+
+import matplotlib.pyplot as plt
+from taipy.gui import Gui, Markdown
+
+
```
Comment (FredLL-Avaiga) [inline doc/gui/examples/charts/matplotlib/image.py]: need to remove one line here

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]

**Review by FredLL-Avaiga — APPROVED** [review]
@jrobinAV Good for HacktoberFest Merge

**Comment by AbisoyeOnanuga** [comment]
I've enjoyed learning through this experience, as well as the incredible support from the Taipy. Thank you guys for pointing out the issues I needed to fix, thereby sharpening my fundamentals of making Pull Requests, including:-
- updating libraries,
- code consistency, 
- filenaming conventions, 
- sorting package imports
- reducing complexity ,etc. 
cheers to @FredLL-Avaiga -- @jrobinAV -- @RymMichaut --@FabienLelaquais for strengthening the developer community and sponsoring Hacktoberfest!
Thanks @SGCODEX for leading the way on this endeavor

**Comment by jrobinAV** [comment]
Thank you, @AbisoyeOnanuga, for your help and contribution.

**Comment by RymMichaut** [comment]
Congratulations @AbisoyeOnanuga 
Could you please contact me at rym.michaut@taipy.io to arrange your shipping :) ?
